executor/write_test.go

```diff
@@ -840,37 +910,10 @@ func (s *testSuite) TestPartitionedTableReplace(c *C) {
 	_, err = tk.Exec(errReplaceSetSQL)
 	c.Assert(err, NotNil)
 	tk.MustExec("rollback")
+}
 
-	tk.MustExec(`drop table if exists replace_test_1`)
-	tk.MustExec(`create table replace_test_1 (id int, c1 int) partition by range (id) (
-			PARTITION p0 VALUES LESS THAN (4),
-			PARTITION p1 VALUES LESS THAN (6),
-			PARTITION p2 VALUES LESS THAN (8),
-			PARTITION p3 VALUES LESS THAN (10),
-			PARTITION p4 VALUES LESS THAN (100))`)
-	tk.MustExec(`replace replace_test_1 select id, c1 from replace_test;`)
-
-	tk.MustExec(`drop table if exists replace_test_2`)
-	tk.MustExec(`create table replace_test_2 (id int, c1 int) partition by range (id) (
-			PARTITION p0 VALUES LESS THAN (10),
-			PARTITION p1 VALUES LESS THAN (50),
-			PARTITION p2 VALUES LESS THAN (100),
-			PARTITION p3 VALUES LESS THAN (300))`)
-	tk.MustExec(`replace replace_test_1 select id, c1 from replace_test union select id * 10, c1 * 10 from replace_test;`)
-
-	errReplaceSelectSQL := `replace replace_test_1 select c1 from replace_test;`
-	tk.MustExec("begin")
-	_, err = tk.Exec(errReplaceSelectSQL)
-	c.Assert(err, NotNil)
-	tk.MustExec("rollback")
-
-	tk.MustExec(`drop table if exists replace_test_3`)
-	replaceUniqueIndexSQL := `create table replace_test_3 (c1 int, c2 int, UNIQUE INDEX (c2)) partition by range (c2) (
-				    PARTITION p0 VALUES LESS THAN (4),
-				    PARTITION p1 VALUES LESS THAN (7),
-				    PARTITION p2 VALUES LESS THAN (11))`
-	tk.MustExec(replaceUniqueIndexSQL)
-	replaceUniqueIndexSQL = `replace into replace_test_3 set c2=8;`
+func testPartitionTableReplace3(tk *testkit.TestKit, c *C) {
```

Comment:
This seems reuse code at the first look, but it's not gool actually.
IMHO, it's a better practice that one test case focus on one thing.
If a test case fail, we can find what's wrong easily, whereas after the rewrite, several cases would fail together and may cost us more time to find the accurate place.